internal/server/middleware.go

```diff
@@ -56,6 +56,37 @@ func DatabaseMiddleware(db *gorm.DB) gin.HandlerFunc {
 	}
 }
 
+func DestinationMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		destinationID := c.GetHeader("Infra-Destination")
+		if destinationID != "" {
+			if db, ok := c.MustGet("db").(*gorm.DB); ok {
+				destination, err := data.GetDestination(db, data.ByOptionalUniqueID(destinationID))
+				if err != nil {
+					if errors.Is(err, internal.ErrNotFound) {
+						c.Next()
+						return
+					}
+
+					sendAPIError(c, err)
+					return
+				}
+
+				// only save if there's significant difference between LastSeenAt and Now
+				if time.Since(destination.LastSeenAt) > time.Second {
```

Comment:
This isn't strictly necessary but the connector tends to send bursts of requests. Seem unreasonable to update the destination for each request in a short window